src/prefect/engine/task_runner.py

```diff
@@ -251,6 +252,11 @@ def run(
                     state, upstream_states=upstream_states
                 )
 
+                # check if any upstream task conditions were not met (and if we need to skip)
+                state = self.check_upstream_conditions_met(
```

Comment:
@cicdw I think there may be an advantage to changing this to a trigger check instead: 
- providing `on_all_conditions_met` and `on_any_conditions_met` triggers (however, no complex combinations would be allowed, only these two in mutually exclusive fashion)
- allowing the downstream receiver task to elect into being conditioned by upstream tasks.

Reply:
After an off-line discussion, I've updated this PR to be enforced by trigger and updated the `all_successful` trigger to account for (enforce) conditions.